fetch/content-type/script.window.js

```diff
@@ -0,0 +1,48 @@
+promise_test(() => {
+  return fetch("resources/script-content-types.json").then(res => res.json()).then(runTests);
+}, "Loading JSON…");
+
+self.isThereABetterWay = undefined;
+
+function runTests(allTestData) {
+  allTestData.forEach(testData => {
+    runScriptTest(testData, false);
+    if (testData.input.length > 1) {
+      runScriptTest(testData, true);
+    }
+  });
+}
+
+function runScriptTest(testData, singleHeader) {
+  async_test(t => {
+    const script = document.createElement("script");
+      t.add_cleanup(() => {
+      script.remove()
+      self.isThereABetterWay = undefined;
+    });
+    script.src = getURL(testData.input, singleHeader);
+    document.head.appendChild(script);
+    if (testData.executes) {
+      script.onload = t.step_func_done(() => {
+        assert_equals(self.isThereABetterWay, testData.encoding === "windows-1252" ? "â‚¬" : "€");
```

Comment:
Is there a reason for checking the encoding here?  Also, I have no idea what all the isThereABetterWay stuff is doing.

Reply:
It's basically making sure the script got decoded correctly. `isThereABetterWay` is a global variable that the script sets with the decoded string value. It's called that way because I was hoping someone would come up with a better mechanism than a global variable. I'll rename it to something more appropriate.